deepspeed/runtime/engine.py

```diff
@@ -2467,6 +2480,12 @@ def load_checkpoint(self,
         ``load_checkpoint()`` wants a pristine model. If insisting to do so, please reinitialize engine
         before ``load_checkpoint()``.
         """
+        checkpoint_engine_tmp = self.checkpoint_engine
+        if not self.config is None and \
+                self._config.nebula_config.enabled and \
+                enable_nebula_load == False:
+            self.checkpoint_engine = CheckpointEngine()
```

Comment:
Can you please explain why `self.checkpoint_engine` is reinitialized here in addition to the initialization in DeepSpeedEngine constructor?

Reply:
Please refer to the above case #1.
customer want to reuse the existing checkpoints saved by deepspeed, then use nebula for saving.

Reply:
This change should also go away once the args are moved to config to allow loading a non-nebula checkpoint, right?

Reply:
Yep, you are right. Removed in the latest commit.